src/atoms/Input/Input.tsx

```diff
@@ -0,0 +1,281 @@
+        : "12px",
+      top: isInputFilled ? "-18px" : "50%",
+      transform: isInputFilled ? "none" : "translateY(-50%)",
+      transition: "all 0.3s ease",
+      fontSize: isInputFilled ? "12px" : "16px",
+      color: isInputFilled ? "gray" : "#a9a9a9",
+      cursor: "text",
+      pointerEvents: "none",
+      whiteSpace: "nowrap",
+    }}
+  >
+    {placeholder}
+  </span>
+);
+
+const TextArea = React.forwardRef<HTMLTextAreaElement, TextAreaProps>(
+  (
+    {
+      state,
+      message,
+      variant = "default",
+      placeholder = "",
+      value = "",
+      margin = "12px 0px",
+      resize = "none",
+      autoHeight = false,
+      useAnimatedPlaceholder = true,
+      ...props
+    },
+    ref
+  ) => {
+    const inputRef = React.useRef<HTMLTextAreaElement | null>(null);
+
+    React.useEffect(() => {
+      if (autoHeight && inputRef.current) {
+        const handleResize = () => {
+          inputRef.current!.style.height = "auto";
+          inputRef.current!.style.height = `${Math.min(
+            inputRef.current!.scrollHeight,
+            250
+          )}px`;
+        };
+        handleResize();
+        inputRef.current.addEventListener("input", handleResize);
+        return () =>
+          inputRef.current?.removeEventListener("input", handleResize);
+      }
+    }, [autoHeight]);
+
+    return (
+      <Wrapper margin={margin}>
+        <InputWrapper>
+          <textarea
+            ref={(current) => {
+              inputRef.current = current;
+              if (typeof ref === "function") ref(current);
+              else if (ref) ref.current = current;
+            }}
+            style={{
+              padding: "10px 12px",
+              width:
+                variant === "halfWidth"
+                  ? "50%"
+                  : variant === "fullWidth"
+                  ? "100%"
+                  : "auto",
+              resize,
+              display: "block",
+              border: "1px solid #e5e5ea",
+              outline: "none",
+              fontSize: "16px",
+              lineHeight: "24px",
+              overflow: autoHeight ? "hidden" : "auto",
+            }}
+            value={value}
+            placeholder="hey!!!"
+            {...props}
+          ></textarea>
+          {useAnimatedPlaceholder && (
+            <AnimatedPlaceholder
+              placeholder={useAnimatedPlaceholder ? "" : placeholder}
+              isInputFilled={!!value}
+              onClick={() => inputRef.current?.focus()}
+              iconWrapperWidth={0}
+            />
+          )}
+        </InputWrapper>
+        {message && (
+          <div
+            style={{
+              padding: "5px",
+              fontSize: "12px",
+              lineHeight: "16px",
+              color: state === "success" ? "green" : "red",
+            }}
+          >
+            {message}
+          </div>
+        )}
+      </Wrapper>
+    );
+  }
+);
+
+export { Input, TextArea };
```

Comment:
no need to export two different components.
instead in input only take a prop named "as" default it will be input. and user can pass as="textarea" then return textarea

Reply:
donnneee